common/reactive/src/main/java/io/helidon/common/reactive/OutputStreamPublisher.java

```diff
@@ -163,6 +195,7 @@ private void complete(Throwable t) {
         subscriber.close(sub -> {
             synchronized (invocationLock) {
                 sub.onError(t);
```

Comment:
Lock doesn't help to prevent rule [§1.7](https://github.com/reactive-streams/reactive-streams-jvm#1.7) violation, we can remove synchronization logic from publisher and wrap subscriber in `io.helidon.common.reactive.SequentialSubscriber`

Reply:
Not a blocker for this PR, just noticing

Reply:
1.7 violation can be seen to not be reachable in this instance.

Reply:
I'm thinking about not very probable use-case of request(-1) vs. exception in publish method

Reply:
You can see `subscriber.close` ensures completion signals will be signalled once and only once, as long as they always go through `complete` which calls `subscriber.close`.

Reply:
Missed that, thanks for clarification

Reply:
Pro tip: when designing routines with CPS, ensure a continuation is invoked _always_. (Not necessarily the same continuation for all outcomes, not necessarily immediately, but you've got to design routines with a guarantee that the continuation is invoked always) You want to not call any continuation only in very rare circumstances.

The problem with the current design of `SingleSubscriberHolder` is that it can omit calling the continuation, and the caller is not told that that is the case. It is the same behaviour as non-termination. The outcome is that `signalCloseComplete` is not guaranteed to be reachable, and the `completionResult.get` can wait indefinitely. The (rather long) timeout then only eventually reclaims the thread.